tests/test_initialization.py

```diff
@@ -1155,6 +1155,106 @@ def test_py_hlvl_parallel_float(tmp_path):
     #       *exactly* two jobs were run in parallel.
 
 
+@pytest.mark.skipif(platform.system() != 'Linux', reason='Parallel support \
+                    only on Linux OS.')
+def test_py_hlvl_parallel_checkpoint(tmp_path):
```

Comment:
this is cool. I was afraid to try parallel checkpointing. Glad to see it seems like it was relatively straightforward to implement?

Reply:
Ya once I added the logic to the preprocessor it worked - so I think the method of handling the checkpoint/netCDF file is pretty reasonable as a consequence.